### ISSUE TITLE
Add ability to remove course

### ISSUE THREAD

**Issue by macrivello** [cpe308-309/polypath#132]

**Comment by macrivello** [comment]
I played around with this and found out we need to change the JPA annotation on the Flowchart class. I wasn't able to get an entity to remove using the API (HTTP DELTE /entity/{id}). The server would return a 200 OK but I could see in the DB that it was not getting deleted. After doing some investigation I found out that we needed to change the CascadeType value in the @OneToMany annotation that establishes the relationship with our flowchart Entry class. Changing it from CascadeType.ALL to CascadeType.REMOVE allows Hibernate to remove the child Entry from the parent Flowchart entry collection when the Child is deleted. 

```
    @OneToMany(targetEntity = Entry.class, mappedBy = "flowchart",
            cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private List<Entry> entries;
```
to 
```
    @OneToMany(targetEntity = Entry.class, mappedBy = "flowchart",
            cascade = CascadeType.REMOVE, fetch = FetchType.EAGER)
    private List<Entry> entries;
```
